src/context/data-context.js

```diff
@@ -46,6 +46,21 @@ const DataProvider = ({ children }) => {
                ...state,
                quizCategories: payload,
             };
+         case "SET_CURRENT_QUESTION_ID":
+            return {
+               ...state,
+               currentQuestionId: payload,
+            };
+         case "SET_CURRENT_CATEGORY":
+            return {
+               ...state,
+               currentCategory: payload,
+            };
+         case "SET_USER_ANSWER":
+            return {
+               ...state,
+               userAnswer: [...state.userAnswer, payload],
+            };
          default:
             return state;
       }
```

Comment:
Suggestion: You can create a separate folder for reducers and add all your reducers there

Reply:
Thanks for the suggestion. Will work on it.